lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,56 @@
+import type { RenovateConfig } from '../../../config/types';
+import { get, getManagerList } from '../../../modules/manager';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig
+  extends Omit<WorkerExtractConfig, 'manager' | 'fileList'> {
+  manager?: string;
+  fileList?: string[];
+}
+
+export function extractFingerprintConfig(
```

Comment:
I don't like that we have a large fingerprint function. Instead:

- We should have a "generate extract config" function, as part of typical app logic
- The result of that function can then be fingerprinted, which should be no more than a few lines

Reply:
Okay converting it into a config narrowing function